File: rust/src/models/price_component_input.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{fee::Fee, product_id::ProductId};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct PriceComponentInput {
    pub fee: Fee,

    pub name: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub product_id: Option<ProductId>,
}

impl PriceComponentInput {
    pub fn new(fee: Fee, name: String) -> Self {
        Self {
            fee,
            name,
            product_id: None,
        }
    }
}

File: typescript/src/models/priceComponentInput.ts
// this file is @generated
import { type Fee, FeeSerializer } from "./fee";
import { type ProductId, ProductIdSerializer } from "./productId";

export interface PriceComponentInput {
  fee: Fee;

  name: string;

  productId?: ProductId | null;
}

export const PriceComponentInputSerializer = {
  _fromJsonObject(object: any): PriceComponentInput {
    return {
      fee: FeeSerializer._fromJsonObject(object["fee"]),
      name: object["name"],
      productId:
        object["product_id"] != null
          ? ProductIdSerializer._fromJsonObject(object["product_id"])
          : undefined,
    };
  },

  _toJsonObject(self: PriceComponentInput): any {
    return {
      fee: FeeSerializer._toJsonObject(self.fee),
      name: self.name,
      product_id:
        self.productId != null
          ? ProductIdSerializer._toJsonObject(self.productId)
          : undefined,
    };
  },
};
